packages/widgets/src/tabbar.ts

```diff
@@ -711,6 +712,44 @@ export class TabBar<T> extends Widget {
    * Handle the `'keydown'` event for the tab bar.
    */
   private _evtKeyDown(event: KeyboardEvent): void {
+    // Allow for navigation using tab key
+    if (event.key === 'Tab') {
+      return;
+    }
+
+    // Check if Enter or Spacebar key has been pressed and open that tab
+    if (
+      event.key === 'Enter' ||
+      event.key === 'Spacebar' ||
+      event.key === ' '
+    ) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      // Get focus element that is in focus by the tab key
+      let focusedElement = document.activeElement;
+
+      if (focusedElement) {
+        if (focusedElement.className === 'lm-TabBar-addButton') {
+          this._addRequested.emit();
+        } else {
+          // Genatate the coordinates for the tab in focus.
+          let tab = focusedElement.getBoundingClientRect();
+          let clientX = tab.left + tab.width / 2;
+          let clientY = tab.top + tab.height / 2;
+
+          // send simulated request with genatated coordinates.
+          let simPointerClick = new PointerEvent('', {
+            button: 0,
+            clientX: clientX,
+            clientY: clientY
+          });
+
+          this._evtPointerDown(simPointerClick);
+          this._evtPointerUp(simPointerClick);
```

Comment:
We discussed this on accessibility meeting today. From implementation side we would prefer this to avoid synthetic events and instead copy (or extract to granular private method) any relevant code, which should be pretty minimal for the purpose of switching tabs - you just need to find the index of the `focusedElement` among tab nodes, update `.currentIndex` and call activate as in [here](https://github.com/jupyterlab/jupyterlab/blob/d98ef15bfa916309c82d105a5d442d915d5dd1df/packages/application/src/shell.ts#L782-L793).

Reply:
@g547315 do you need help addressing the above comment?

Reply:
Code changes
Removed synthetic events onclick event creation
Added helper method to find all adjecent child nodes 
Added helper method to find index of the focusedElement from adjecent child nodes
Added functionality to update currentIndex


User-facing changes
Elements on left and right TabBar and DockPanel TabBar now respond to enter or spacebar

Backwards-incompatible changes
None